packages/hooks/src/createHasHook.js

```diff
@@ -5,17 +5,24 @@
  * @param  {Object}   hooks Stored hooks, keyed by hook name.
  *
  * @return {Function}       Function that returns whether any handlers are
- *                          attached to a particular hook.
+ *                          attached to a particular hook and optional namespace.
  */
 function createHasHook( hooks ) {
 	/**
-	 * Returns how many handlers are attached for the given hook.
+	 * Returns whether any handlers are attached for the given hookName and optional namespace.
 	 *
-	 * @param  {string}  hookName The name of the hook to check for.
+	 * @param {string} hookName  The name of the hook to check for.
+	 * @param {string} namespace Optional. The unique namespace identifying the callback in the form `vendor/plugin/function`.
 	 *
 	 * @return {boolean} Whether there are handlers that are attached to the given hook.
 	 */
-	return function hasHook( hookName ) {
+	return function hasHook( hookName, namespace ) {
+		// Use the namespace if provided.
+		if ( 'undefined' !== typeof namespace ){
+			return hookName in hooks &&
+				hooks[ hookName ].handlers.some( hook => hook.namespace === namespace );
+		}
+
 		return hookName in hooks;
```

Comment:
Minor: Regardless of whether `namespace` is provided, we have this same condition apply, so rather than duplicate, we could move it as a primary check of the function:

```js
if ( ! ( hookName in hooks ) ) {
	return false;
}

if ( 'undefined' === typeof namespace ) {
	return true;
}

return hooks[ hookName ].handlers.some( ( hook ) => {
	return hook.namespace === namespace;
} );
```